testplan/common/utils/networking.py

```diff
@@ -8,10 +8,65 @@
     absolute_import,
 )
 import socket
+import ipaddress
 
 
 def port_to_int(port):
+    """Convert a port string to an integer."""
     try:
         return int(port)
     except ValueError:
         return socket.getservbyname(port)
+
+
+def format_access_urls(host, port, url_path):
+    """
+    Format a string to log to give HTTP access to a URL endpoint listening
+    on a particular host/port. Handles special 0.0.0.0 IP and formats IPV6
+    addresses. The returned string is indented by 4 spaces.
+
+    :param host: hostname or IP address
+    :type host: ``str``
+    :param port: port number
+    :type port: ``int``
+    :param url_path: path to format after host:port part of URL (expected to
+        contain leading "/" e.g "/index.html")
```

Comment:
It's good this is mentioned in the docstring but isn't it trivial to just add this check?

```py
path = path.strip()
path = ('/' if not path.startswith('/') else '') + path
```

Reply:
yeah it doesn't hurt to add a check